car.go

```diff
@@ -197,60 +200,78 @@ type batchStore interface {
 }
 
 func LoadCar(s Store, r io.Reader) (*CarHeader, error) {
+	logger.Info("will load car now")
 	cr, err := NewCarReader(r)
 	if err != nil {
 		return nil, err
 	}
 
 	if bs, ok := s.(batchStore); ok {
+		logger.Info("will load car fast")
 		return loadCarFast(bs, cr)
 	}
 
+	logger.Info("will load car slow")
 	return loadCarSlow(s, cr)
 }
 
 func loadCarFast(s batchStore, cr *CarReader) (*CarHeader, error) {
+	nBlocks := 0
 	var buf []blocks.Block
 	for {
 		blk, err := cr.Next()
 		switch err {
 		case io.EOF:
 			if len(buf) > 0 {
 				if err := s.PutMany(buf); err != nil {
+					logger.Errorf("failed to write %d blocks on completion, err=%s, nBlocksWritten=%d", len(buf), err, nBlocks)
```

Comment:
I'd suggest putting the error last, so that if the error message is really long it's easier to read the other stuff (eg nBlocksWritten)

Reply:
Done.